harvdev_utils/general_functions/count_value_frequency.py

```diff
@@ -0,0 +1,33 @@
+"""
+.. module:: count_value_frequency
+   :synopsis: Logs the frequency of all values in a list.
+
+.. moduleauthor:: Gil dos Santos dossantos@morgan.harvard.edu
+"""
+
+import logging
+
+log = logging.getLogger(__name__)
+
+
+def count_value_frequency(input_list):
+    """
+    A function that logs frequency of values in a list.
+    Args:
+        A list.
+    Returns:
+        Logs value frequency, no return per se.
+    Raises:
+        None.
+    """
+
+    unique_values = set(input_list)
```

Comment:
This is normally done as :- 
 for element in input_list:
        if element in count_dict:
            count_dict[element] += 1
        else:
           count_dict[element] = 1

The above seems faster as we do not have to create a set that is used just once to set initial values. Could be wrong, very curious now as to what the speed effects are. As your version does not need an if statement in the loop? hmmm

Reply:
I think @gildossantos _might_ be faster because he's not doing a lookup for every value (`if element in`). Using `set` to remove duplicates in order to get a unique list of keys is pretty neat.

Reply:
And then it's just 3 loops to get the answer, regardless of list size, instead of `n` loops (`if element in`).

Reply:
Did say i could be wrong. I may have to generate some tests and do some profiling as my curiousity has been tweeked now.

Reply:
Hahaha, do it. Sorry, didn't mean to be critical, just genuinely curious. I haven't seen an approach like @gildossantos before, definitely stealing it for my unique counts.

Reply:
The two approaches are microseconds apart for list of 200,000 strings (5 types), so keeping as is.